org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3329,19 +3569,20 @@ class WorkspaceEdit {
     	this.changes = changes
     }
     
-    new(List<TextDocumentEdit> documentChanges) {
-    	this.documentChanges = documentChanges
-    }
-    
-    /**
-     * @deprecated According to the protocol documentation, it doesn't make sense to send both
-     * 		changes and documentChanges
-     */
-    @Deprecated
-    new(Map<String, List<TextEdit>> changes, List<TextDocumentEdit> documentChanges) {
-    	this.changes = changes
-    	this.documentChanges = documentChanges
-    }
+
+	new(List<Either<TextDocumentEdit, ResourceOperation>> documentChanges) {
+		this.documentChanges = documentChanges
+	}
+
+	/**
+	 * @deprecated According to the protocol documentation, it doesn't make sense to send both
+	 * 		changes and documentChanges
+	 */
+	@Deprecated
+	new(Map<String, List<TextEdit>> changes, List<Either<TextDocumentEdit, ResourceOperation>> documentChanges) {
```

Comment:
You can safely remove this deprecated constructor because the API has been broken anyway.